polus-ftl-label-plugin/src/ftl.pyx

```diff
@@ -0,0 +1,966 @@
+    cdef np.ndarray start_ind = np.zeros(rle_objects.shape[0],dtype=np.uint64)
+    
+    for i in range(ndims):
+        start_ind += rle_objects[:,i] 
+        start_ind *= image_shape[i+1]
+    start_ind += rle_objects[:,obj_start]
+
+    cdef unsigned long [:] start_ind_memview = start_ind
+    for i in range(labels.shape[0]):
+        fill_n(&linear_image[start_ind_memview[i]],
+                rle_objects[i,obj_end] - rle_objects[i,obj_start],
+                labels[i])
+    
+    return label_image
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.initializedcheck(False)
+cdef np.ndarray generate_output_16(unsigned int [:,:] rle_objects,
+                                  unsigned long [:] labels,
+                                  tuple image_shape):
+    """ generate_output_16 Generate 16-bit output image
+
+    This method takes an rle compressed set of data and generates
+    an image with up to 2**16-1 unique labels.
+
+    Args:
+        rle_objects (np.ndarray): 2d array containg rle indices
+        labels (np.ndarray): 1d array of labels for each object
+        image_shape (tuple): Shape of the output image
+
+    Outputs:
+        np.ndarray: 16-bit labeled image
+    
+    """
+    
+    # Initialize iteration counter
+    cdef long i
+
+    # Initialize the output
+    # print('Initializing the output...')
+    cdef np.ndarray label_image = np.zeros(image_shape,dtype=np.uint16)
+    cdef unsigned short [:] linear_image = label_image.reshape(-1)
+    # print('label_image.shape={}'.format(image_shape))
+
+    # Get the indices
+    # print('Initializing the indices...')
+    cdef long long ndims = len(image_shape) - 1
+    cdef long long obj_start = rle_objects.shape[1] - 2
+    cdef long long obj_end = rle_objects.shape[1] - 1
+    cdef np.ndarray start_ind = np.zeros(rle_objects.shape[0],dtype=np.uint64)
+    
+    # print('Calculating the indices...')
+    for i in range(ndims):
+        start_ind += rle_objects[:,i] 
+        start_ind *= image_shape[i+1]
+    start_ind += rle_objects[:,obj_start]
+    # print(start_ind[:10])
+
+    # print('Generating the output...')
+    cdef unsigned long [:] start_ind_memview = start_ind
+    for i in range(labels.shape[0]):
+        # print('i={}'.format(i))
+        # print('start_ind_memview={}'.format(start_ind_memview[i]))
+        # print('size={}'.format(rle_objects[i,obj_end] - rle_objects[i,obj_start]))
+        # print('value={}'.format(labels[i]))
+        fill_n(&linear_image[start_ind_memview[i]],
+                rle_objects[i,obj_end] - rle_objects[i,obj_start],
+                labels[i])
+    
+    return label_image
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.initializedcheck(False)
+cdef np.ndarray generate_output_32(unsigned int [:,:] rle_objects,
+                                   unsigned long [:] labels,
+                                   tuple image_shape):
+    """ generate_output_32 Generate 32-bit output image
```

Comment:
I assume you check label count and call appropriate one of these depending on total object count somewhere below. But, why stop at 32 bit? Why not also have a double precision? Just curious.

Reply:
My general thought is that we won't have more than `2^32-1` objects. For now that's probably a safe assumption. I don't think we would ever want double precision, but we might want 64-bit integers.